lightning/src/ln/channelmanager.rs

```diff
@@ -12881,6 +12887,7 @@ impl Readable for HTLCFailureMsg {
 					channel_id: Readable::read(reader)?,
 					htlc_id: Readable::read(reader)?,
 					reason: Readable::read(reader)?,
+					attribution_data: Readable::read(reader)?,
```

Comment:
change write side to use version 2/3 and it will have the length descriptor. can add additional content at the end.